io/router/router.go

```diff
@@ -182,6 +182,19 @@ func (q *Router) Queue(events ...event.Event) bool {
 			}
 		case clipboard.Event:
 			q.cqueue.Push(e, &q.handlers)
+		case transfer.DataEvent:
+			pq := &q.pointer.queue
+			for _, h := range pq.hitTree {
+				if h.tag != nil {
+					targetMimes := pq.handlers[h.tag].targetMimes
+					for _, mimeType := range targetMimes {
+						if mimeType == e.Type {
+							q.handlers.Add(h.tag, e)
+							break
+						}
+					}
+				}
+			}
 		}
```

Comment:
This is probably correct, but seems to duplicate existing logic in `pointerQueue`. Can this code be merged with, say, `pointerQueue.deliverTransferDataEvent` so the internal and external code dnd paths become similar?